### PR TITLE
Feature: add customization settings

### PR DESCRIPTION
**Pull request by oyershov**

### CODE REVIEW

File: src/containers/TradingChart/index.tsx [hunk 1]
```diff
@@ -174,6 +180,24 @@ export class TradingChartComponent extends React.PureComponent<Props> {
...
+        if (this.tvWidget && currentMarket) {
+            try {
+                this.tvWidget.remove();
+            } catch (error) {
+                window.console.log(`TradingChart unmount failed (Rebuild chart): ${error}`);
+            }
+
+            this.setChart(markets, currentMarket, colorTheme);
+        }
+    }
```
Comment (akhlopiachyi) [inline src/containers/TradingChart/index.tsx:199]: add semicolon

File: src/containers/TradingChart/index.tsx [hunk 2]
```diff
@@ -174,6 +180,24 @@ export class TradingChartComponent extends React.PureComponent<Props> {
...
+            currentMarket,
+            markets,
+        } = this.props;
+
+        if (this.tvWidget && currentMarket) {
+            try {
+                this.tvWidget.remove();
+            } catch (error) {
+                window.console.log(`TradingChart unmount failed (Rebuild chart): ${error}`);
+            }
```
Comment (akhlopiachyi) [inline src/containers/TradingChart/index.tsx:195]: use finally

File: src/containers/TradingChart/index.tsx [hunk 3]
```diff
@@ -174,6 +180,24 @@ export class TradingChartComponent extends React.PureComponent<Props> {
...
+
+            this.setChart(markets, currentMarket, colorTheme);
+        }
+    }
+
     private languageIncluded (lang: string ) {
```
Comment (akhlopiachyi) [inline src/containers/TradingChart/index.tsx:201]: private languageIncluded = (lang: string ) => {

File: src/custom/tradingChartConfig.ts [hunk 4]
```diff
@@ -1,10 +1,12 @@
 /* eslint-disable */
 import { ThemeName } from '../charting_library/charting_library.min';
 import { colors } from '../constants';
+import { convertRgbToHex, getStylesValueByKey } from '../helpers';
 
 export const customWidgetParams = {};
 
 export const customWidgetOptions = (colorTheme?: string) => {
+
```
Comment (akhlopiachyi) [inline src/custom/tradingChartConfig.ts:9]: remove empty line

File: src/modules/public/customization/reducer.test.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
+import * as actions from './actions';
+import { customizationReducer, initialCustomizationState } from './reducer';
+import { CustomizationDataInterface } from './types';
+
+describe('customizationReducer', () => {
+    const fakeCustomization: CustomizationDataInterface = {
+        settings: '{\"theme_id\": \"1\",\"theme_colors\":[]}',
+    };
+
```
Comment (akhlopiachyi) [inline src/modules/public/customization/reducer.test.ts:9]: remove empty line

File: src/modules/public/customization/sagas/customizationFetchSaga.test.ts [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+import { rootSaga } from '../../..';
+import { mockNetworkError, setupMockAxios, setupMockStore } from '../../../../helpers/jest';
+import {
+    customizationData,
+    customizationError,
+    customizationFetch,
+} from '../actions';
+import { CustomizationDataInterface } from '../types';
+
+// tslint:disable no-any no-magic-numbers
```
Comment (akhlopiachyi) [inline src/modules/public/customization/sagas/customizationFetchSaga.test.ts:13]: remove

File: src/modules/user/customization/reducer.test.ts [hunk 7]
```diff
@@ -0,0 +1,47 @@
+import { CustomizationDataInterface } from '../../public/customization';
+import * as actions from './actions';
+import { customizationUpdateReducer, initialCustomizationUpdateState } from './reducer';
+
+describe('customizationReducer', () => {
+    const fakeCustomization: CustomizationDataInterface = {
+        settings: '{\"theme_id\": \"1\",\"theme_colors\":[]}',
+    };
+
```
Comment (akhlopiachyi) [inline src/modules/user/customization/reducer.test.ts:9]: remove empty line

File: src/modules/user/customization/sagas/customizationUpdateSaga.test.ts [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+import { rootSaga } from '../../..';
+import { mockNetworkError, setupMockAxios, setupMockStore } from '../../../../helpers/jest';
+import { CustomizationDataInterface } from '../../../public/customization';
+import {
+    customizationUpdate,
+    customizationUpdateData,
+    customizationUpdateError,
+} from '../actions';
+
+// tslint:disable no-any no-magic-numbers
```
Comment (akhlopiachyi) [inline src/modules/user/customization/sagas/customizationUpdateSaga.test.ts:13]: remove

File: src/modules/user/customization/sagas/customizationUpdateSaga.ts [hunk 9]
```diff
@@ -0,0 +1,25 @@
+// tslint:disable-next-line
```
Comment (akhlopiachyi) [inline src/modules/user/customization/sagas/customizationUpdateSaga.ts:1]: remove

File: src/modules/user/customization/sagas/index.ts [hunk 10]
```diff
@@ -0,0 +1,8 @@
+// tslint:disable-next-line
```
Comment (akhlopiachyi) [inline src/modules/user/customization/sagas/index.ts:1]: remove

File: src/themes/index.ts [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+    theme: ThemeColorInterface[];
+}
+
+export const AVAILABLE_COLOR_THEMES: ThemeInterface[] = [
+    { id: 0, title: 'page.body.customization.themes.theme.darkBlue.title', theme: DarkBlueTheme },
+    { id: 1, title: 'page.body.customization.themes.theme.darkRed.title', theme: DarkRedTheme },
+    { id: 2, title: 'page.body.customization.themes.theme.purple.title', theme: PurpleTheme },
+    { id: 3, title: 'page.body.customization.themes.theme.green.title', theme: GreenTheme },
+];
+
```
Comment (akhlopiachyi) [inline src/themes/index.ts:24]: remove empty line

File: src/components/Customization/Themes/ColorSettings/index.tsx [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+import { handleConvertColorCode } from '../index';
+
+interface OwnProps {
+    handleCloseColorSettings: () => void;
+    item: ThemeColorTitleInterface;
+    translate: (key: string) => string;
+    handleTriggerChartRebuild?: () => void;
+}
+
+interface State {
```
[the remark this answers is not in the document]
Comment (oyershov) [reply src/components/Customization/Themes/ColorSettings/index.tsx:15]: no, I separated data-fetching and rendering concerns. It`s main point.

**Pull request merged**